### ISSUE TITLE
Validator proposal: same_as

### ISSUE THREAD

**Issue by glaucocustodio** [ruby-grape/grape#1846]
Hi folks, I just added a custom validator (`same_as`) to my app and I would like to know whether it would be welcome to grape itself:

```ruby
resources :passwords do
  params do
    requires :current_password, allow_blank: false
    requires :password, allow_blank: false
    requires :password_confirmation, allow_blank: false, same_as: :password
  end
  post do
  end
end
```

Thanks.

**Comment by myxoh** [comment]
Can you think of any other example in which it would be useful other that the `password` `password_confirmation`? 
I've spent some time trying to think of alternative use cases and I'm having a hard time. If there aren't any it seems to be as a bit of an overkill to have a validator for just one use case, especially as there are plenty of libraries that handle this specific validation at the model level (arguably it'd be better at a param level, but seems to niche)

I'll let @dblock make the final call

**Comment by glaucocustodio** [comment]
I can only think in confirmation cases like email and password. Anyway I would say it is worth to have common validators built-in like this one and some others like `length`. Those are very small, isolated (one file) and nice to have.

**Comment by glaucocustodio** [comment]
I am closing because it was implemented on PR #1850.
